src/Castle.Core/NullStateStaticAnalysisAttributes.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright 2004-2023 Castle Project - http://www.castleproject.org/
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace System.Diagnostics.CodeAnalysis
```

Comment:
Is there a better directory to put this polyfill attribute file under than the root `src/Castle.Core/`?

Reply:
Done. I guess the exact directory name doesn't matter all that much, I went with `Compatiblity` for now.